src/dex/uniswap-v2/uniswap-v2.ts

```diff
@@ -537,32 +540,56 @@ export class UniswapV2
         return null;
 
       await this.batchCatchUpPairs([[from, to]], blockNumber);
-
-      const pairParam = await this.getPairOrderedParams(from, to, blockNumber);
+      const isSell = side === SwapSide.SELL;
+      const pairParam = await this.getPairOrderedParams(
+        from,
+        to,
+        blockNumber,
+        transferFees.srcDexFee,
+      );
 
       if (!pairParam) return null;
 
-      const unitAmount = getBigIntPow(
-        side == SwapSide.BUY ? to.decimals : from.decimals,
+      const unitAmount = getBigIntPow(isSell ? from.decimals : to.decimals);
+
+      const [unitWithFee, ...amountsWithFee] = applyTransferFee(
+        [unitAmount, ...amounts],
+        side,
+        transferFees.srcFee,
+        SRC_TOKEN_PARASWAP_TRANSFERS,
       );
-      const unit =
-        side == SwapSide.BUY
-          ? await this.getBuyPricePath(unitAmount, [pairParam])
-          : await this.getSellPricePath(unitAmount, [pairParam]);
-
-      const prices =
-        side == SwapSide.BUY
-          ? await Promise.all(
-              amounts.map(amount => this.getBuyPricePath(amount, [pairParam])),
-            )
-          : await Promise.all(
-              amounts.map(amount => this.getSellPricePath(amount, [pairParam])),
-            );
 
+      const unit = isSell
+        ? await this.getSellPricePath(unitWithFee, [pairParam])
+        : await this.getBuyPricePath(unitWithFee, [pairParam]);
+
+      const prices = isSell
+        ? await Promise.all(
+            amountsWithFee.map(amount =>
+              this.getSellPricePath(amount, [pairParam]),
+            ),
+          )
+        : await Promise.all(
+            amountsWithFee.map(amount =>
+              this.getBuyPricePath(amount, [pairParam]),
+            ),
+          );
+
+      const outputsWithFee = applyTransferFee(
+        applyTransferFee(
+          prices,
+          side,
+          isSell ? transferFees.destFee : transferFees.srcFee,
+          isSell ? DEST_TOKEN_PARASWAP_TRANSFERS : SRC_TOKEN_PARASWAP_TRANSFERS,
+        ),
+        side,
+        isSell ? transferFees.destDexFee : transferFees.srcDexFee,
```

Comment:
`srcDexFee` should not be applied as it is included in the pool fee accounted for in pricing.

Reply:
You are right, replaced with 0